### PR TITLE
implement 'categories' method

### LINKED ISSUE #178: add a test to keep constants up to date
The test should request and scrape the Google Play website to obtain the current list of app categories and make sure they are the same as those in the constants file.

This way whenever they become outdated the tests will start to fail.

### PR DESCRIPTION
**Pull request by nonphp**
- extract list of categories from dropdown menu
- categories tests

Fixes #178

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,11 +5,13 @@ const constants = require('./lib/constants');
 const memoizee = require('memoizee');
 
 const appMethod = require('./lib/app');
+const categoriesMethod = require('./lib/categories');
 const getParseList = require('./lib/utils/parseList');
 const parseList = R.partial(getParseList, [appMethod]);
 
 const methods = {
   app: appMethod,
+  categories: categoriesMethod,
```
Comment (facundoolano) [inline index.js]: Put the require directly here instead of defining a variable, to follow the convention as the rest of the methods.

Comment (facundoolano) [reply index.js]: categories should also be defined in the memoized object below.

File: test/lib.categories.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+describe('Categories method', () => {
+  it('should fetch valid list of categories', () => {
+    return gplay.categories().then(categoryIds => {
+      assert.isArray(categoryIds);
+      assert.isTrue(categoryIds.length > 0);
+    });
+  });
+
+  it('should have all categories from constant list of categories', () => {
+    return gplay.categories().then(categoryIds => {
```
Comment (facundoolano) [inline test/lib.categories.js]: It should also test the other way around: all categories should have their constant.

Suggestion: use [R.difference](https://ramdajs.com/docs/#difference) to do both checks with a set operation.

Comment (nonphp) [reply test/lib.categories.js]: Hi. What about categories that are not present in dropdown menu? -- APPLICATION and GAME
Maybe add them into resulting array as constants?

Comment (facundoolano) [reply test/lib.categories.js]: Yeah, that sounds reasonable

Comment (nonphp) [reply test/lib.categories.js]: Done

**Pull request merged**